org.eclipse.draw2d/src/org/eclipse/draw2d/geometry/Rectangle.java

```diff
@@ -399,6 +399,22 @@ public Rectangle getIntersection(Rectangle rect) {
 		return getCopy().intersect(rect);
 	}
 
+	/**
+	 * Returns a new Interval based on the width or height of this Rectangle. If
+	 * {@code isHorizontal} is true, a new Interval beginning at the x coordinate
+	 * with the width as length is returned. Otherwise a new Interval beginning at
+	 * the y coordinate with the height as length is returned.
+	 * 
+	 * @param isHorizontal {@code true} for the horizontal Interval, {@code false}
+	 *                     for the vertical.
+	 * @return the {@link Interval} of projection on X or Y axis, depending on
+	 *         <code>isHorizonal</code> param.
+	 * @since 3.13
+	 */
+	public Interval getInterval(boolean isHorizontal) {
```

Comment:
not sure where you need it but would a getHorInterval and getVerInterval be better readable in client code?

Reply:
I agree. Calling `getHorInterval()` looks a lot more descriptive than `getInterval(true)`